spring-cloud-starter-task-composedtaskrunner/src/main/java/org/springframework/cloud/task/app/composedtaskrunner/TaskLauncherTasklet.java

```diff
@@ -147,7 +147,7 @@ public RepeatStatus execute(StepContribution contribution,
 			TaskExecution taskExecution =
 					this.taskExplorer.getTaskExecution(this.executionId);
 			if (taskExecution != null && taskExecution.getEndTime() != null) {
-				if (taskExecution.getExitCode() != 0) {
+				if (taskExecution.getExitCode() == null || taskExecution.getExitCode() != 0) {
 					throw new UnexpectedJobExecutionException("Task returned a non zero exit code.");
```

Comment:
@cppwfs Sorry about the delay in reviewing this. 
We can probably throw different exception message when the exit code is `null` as it may help to debug it easier.